Make the list navigation buttons work

DaniElectra opened the pull request:
When a user interacts with the Previous or Next button, it will show the next or previous page of the list, or rolls to the beginning or end if the user tries to go out of bounds.

Note that the bot will reply to the message with the buttons (in other words, its own message). It will also use the list of the user who is interacting with the buttons, not the user who originally triggered the list command. I'm not sure how to fix these issues though since I don't know if it's possible or how to get the ID of the user who initially triggered the list, which would allow solving these issues.

Example:

![image](https://github.com/user-attachments/assets/b98f40cc-4aa3-4911-9ea5-6afaa0b40416)

DaniElectra committed:
Make the list navigation buttons work

When a user interacts with the Previous or Next button, it will show the
next or previous page of the list, or will show an error if the user
tries to go out of bounds.

Note that the bot will reply to the message with the buttons (in other
words, its own message). It will also use the list of the user who is
interacting with the buttons, not the user who originally triggered the
list command.

DaniElectra committed:
Roll to the beginning or end if out of bounds

OFFTKP commented:
Thanks for the pr!

When we set the id of the button, could we include the user id of the person that did the /list? If so when a button is pressed we could extract this user id from the button id string like we do with the page number.

Replying to itself (instead of editing the message) is a bit meh although it's better than not working buttons for sure.

DaniElectra commented:
> Thanks for the pr!
> 
> When we set the id of the button, could we include the user id of the person that did the /list? If so when a button is pressed we could extract this user id from the button id string like we do with the page number.

That seems reasonable. It would break already existing buttons, but who is going to use buttons from previous messages anyway, so I'll do that.

> Replying to itself (instead of editing the message) is a bit meh although it's better than not working buttons for sure.

I have tried the functions for editing the message (`edit_response` and `edit_original_response`) but they wouldn't work, so `reply` it is.

DaniElectra committed:
Store the user ID in the list navigation buttons

This allows mapping the buttons to the user who triggered the list
command. Also edit the existing list instead of making a new one with
the new entries.

DaniElectra commented:
I added the user ID to the buttons. I also made the bot edit the existing list using `dpp::ir_update_message` on the `reply`

wheremyfoodat commented:
So pog
